cedar-policy/src/api.rs

```diff
@@ -440,11 +441,12 @@ impl ResidualResponse {
 /// Errors encountered while evaluating policies or expressions, or making
 /// authorization decisions.
 #[derive(Debug, Clone, PartialEq, Eq, Error)]
-pub enum EvaluationError {
+pub enum ApiEvaluationError {
```

Comment:
I'm not sure this name change makes things clearer for users.  I think we should endeavor to make the public type called `EvaluationError`, regardless of what gymnastics that requires under the hood here.  (Probably means we can't `use` Core's `EvaluationError` unqualified in this file, and have to refer to it as something like `core::EvaluationError`.)